AdrianBarros/assignments/slidingwindow/lc209/lc209.py

```diff
@@ -1,45 +1,62 @@
 import math
 
+# inputs: int= s arr=[]
+# output: int=n
+# loop throu the arr
+# window start, window end
+# des> i = slide the window
+# until the sum > des
+# min num_i
+
 
 def smallest_subarray_with_given_sum(s, arr):
+    min_arr = float('inf')
+    cur_arr, win_start, curr_val = 0
+
+    # Step 1 Loop through the array
+    for i in range(len(arr)):
+        # step 2: add the value to the current sum and extend the size of the curr subarray
+        curr_val += arr[i]
+        cur_arr += 1
+        # step 3 check if the value is what we are looking for
+        if curr_val >= s:
+            # step 4 update the min subarray value
+            min_arr = min(min_arr, cur_arr)
+            # Step 5 update the sub and value
+            curr_val -= arr[win_start]
+            win_start += 1
+            cur_arr -= 1
+
+    return min_arr
```

Comment:
I like the comments sprinkled throughout! Good code structure and cleanliness. Remember if you gotta be nice and type it twice! #Reps #DoItAgain